### PR TITLE
Don't purge fail queues before reindex (#1422)

### PR DESCRIPTION
**Pull request by danielsotirhos**

### CODE REVIEW

File: scripts/reindex.py [hunk 1]
```diff
@@ -69,7 +69,11 @@ def __init__(self, prog) -> None:
 parser.add_argument('--purge',
                     default=False,
                     action='store_true',
-                    help='Purge the queues before taking any action on the index.')
+                    help='Purge all queues except the fail queues before taking any action on the index.')
+parser.add_argument('--purge-all',
```
Comment (hannes-ucsc) [inline scripts/reindex.py]: No need to retain existing functionality. Please remove.

File: scripts/reindex.py [hunk 2]
```diff
@@ -93,21 +97,30 @@ def main(argv: List[str]):
...
-    if args.purge:
+    if args.purge or args.purge_all:
         queue_manager = Queues()
         queues = dict(queue_manager.azul_queues())
-        logger.info('Disabling lambdas ...')
-        queue_manager.manage_lambdas(queues, enable=False)
-        logger.info('Purging queues ...')
-        queue_manager.purge_queues_unsafely(queues)
+        if not args.purge_all:
+            queues = {key: val for key, val in queues.items() if '-fail-' not in key}
```
Comment (hannes-ucsc) [inline scripts/reindex.py]: Use this instead:

```diff
Index: src/azul/__init__.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/azul/__init__.py	(revision f07e177b45552345a833cd09f5504e788b001390)
+++ src/azul/__init__.py	(date 1573677194710)
@@ -474,8 +474,15 @@
 
     @property
     def all_queue_names(self):
-        return (self.token_queue_name, self.document_queue_name, self.fail_queue_name,
-                self.fail_fifo_queue_name, self.notify_queue_name)
+        return self.work_queue_names + self.fail_queue_names
+
+    @property
+    def fail_queue_names(self):
+        return self.fail_fifo_queue_name, self.fail_queue_name
+
+    @property
+    def work_queue_names(self):
+        return self.notify_queue_name, self.token_queue_name, self.document_queue_name
 
     manifest_lambda_basename = 'manifest'
 
```

**Comment by hannes-ucsc** [comment]
See https://github.com/DataBiosphere/azul/issues/1447 and https://github.com/DataBiosphere/azul/issues/1448

File: scripts/reindex.py [hunk 3]
```diff
@@ -96,17 +96,18 @@ def main(argv: List[str]):
     if args.purge:
         queue_manager = Queues()
         queues = dict(queue_manager.azul_queues())
+        for fail_queue_name in config.fail_queue_names:
```
Comment (hannes-ucsc) [inline scripts/reindex.py]: Can we invert the logic and use `config.work_queue_names`?

**Comment by hannes-ucsc** [comment]
Sorry for the fragmented review, but technically, the PR was still assigned to me which should have precluded you from pushing to it.

**Comment by danielsotirhos** [comment]
> Sorry for the fragmented review, but technically, the PR was still assigned to me which should have precluded you from pushing to it.

Yes, sorry. Saw your comment via email and jumped back on the branch to make changes without checking the ticket on github first.

File: scripts/reindex.py [hunk 4]
```diff
@@ -96,9 +96,10 @@ def main(argv: List[str]):
     if args.purge:
         queue_manager = Queues()
         queues = dict(queue_manager.azul_queues())
+        queues = {key: val for key, val in queues.items() if key in config.work_queue_names}
```
Comment (hannes-ucsc) [inline scripts/reindex.py]:
```suggestion
        # FIXME: eliminate need for filtering (https://github.com/DataBiosphere/azul/issues/1448)
        queues = {k: v for k, v in queues.items() if k in config.work_queue_names}
```

AFAICT, there's precedence for using `k` and `v` all over the codebase. Please stick with existing conventions as outlined in the contributions guide. Let me know if I am wrong about that.

Comment (danielsotirhos) [reply scripts/reindex.py]: No you are correct. I had thought you previously recommended the opposite but was mistaking a different situation that doesn't apply here inside a dict comprehension.

**Pull request merged**